net.go

```diff
@@ -229,9 +223,11 @@ func Listen(laddr ma.Multiaddr) (Listener, error) {
 
 	var nl net.Listener
 	switch lnet {
-	case "utp":
-		// nl, err = utp.Listen(lnet, lnaddr)
-		return nil, fmt.Errorf("utp is currently broken")
+	case "utp", "utp4", "utp6":
+		s, nserr := utp.NewSocket("udp"+lnet[3:], lnaddr)
+		err = nserr
+		nl = &mautp.Listener{s}
```

Comment:
Put thins in a `Listen` call in our wrapper package. multiaddr-net shouldn't have to know all this anacrolix/utp specific stuff.